From 241d5e68a91be90dd8f81b6ed212f39b05c50346 Mon Sep 17 00:00:00 2001
From: Suman Mukherjee <sumam@codeaurora.org>
Date: Tue, 16 Dec 2014 15:50:15 +0530
Subject: [PATCH 135/192] Camera: Set preview fps after recording.

4k DCI recorded FPS is getting picked up from profile and not
getting reset to preview FPS after stop recording.
Fix made to set the preview FPS range after stop recording.

Change-Id: I4892e6fcee3eefbcbd6ca9426e56e2303cdad844
---
 src/com/android/camera/VideoModule.java | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index 0fdb858a1..b6f1531af 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -2259,6 +2259,18 @@ public class VideoModule implements CameraModule,
                 mUI.hideSurfaceView();
                 // Switch back to use SurfaceTexture for preview.
                 startPreview();
+            } else {
+                if (is4KEnabled()) {
+                    int[] fpsRange = CameraUtil.getMaxPreviewFpsRange(mParameters);
+                    if (fpsRange.length > 0) {
+                        mParameters.setPreviewFpsRange(
+                                fpsRange[Parameters.PREVIEW_FPS_MIN_INDEX],
+                                fpsRange[Parameters.PREVIEW_FPS_MAX_INDEX]);
+                    } else {
+                        mParameters.setPreviewFrameRate(30);
+                    }
+                    mCameraDevice.setParameters(mParameters);
+                }
             }
         }
         // Update the parameters here because the parameters might have been altered
-- 
2.17.1

